biogeochem/EDCanopyStructureMod.F90

```diff
@@ -2010,15 +2007,26 @@ subroutine update_hlm_dynamics(nsites,sites,fcolumn,bc_out)
            end do
            
         endif
+
+        ! If running hydro, perform a final check to make sure that we
+        ! have conserved water. Since this is the very end of the dynamics
+        ! cycle. No water should had been added or lost to the site during dynamics.
+        ! With growth and death, we may have shuffled it around.
+        ! For recruitment, we initialized their water, but flagged them
+        ! to not be included in the site level balance yet, for they
+        ! will demand the water for their initialization on the first hydraulics time-step
+        
+        if (hlm_use_planthydro.eq.itrue) then
+           call UpdateH2OVeg(sites(s),bc_out(s),bc_out(s)%plant_stored_h2o_si,1)
+        end if
         
      end do
 
-     ! If hydraulics is turned on, update the amount of water bound in vegetation
+     ! This call is purely for diagnostics and history variables
      if (hlm_use_planthydro.eq.itrue) then
         call RecruitWaterStorage(nsites,sites,bc_out)
```

Comment:
@xuchongang , thanks for bringing this up.  After re-reading this, I'm realizing that we are not accounting for the water in the new recruits in the total mass balance. 
Did we once have code that pulled water from the soil and put it in the new-recruits, or have we thus far always treated it as an error term?

Reply:
It is in subroutine RecruitWUptake(nsites,sites,bc_in,dtime,recruitflag)

Reply:
Ok, I see it now. I think we are accounting for the mass correctly then.   That routine is called inside the main hydraulics loop, and should only be relevant on the first call following a dynamics call. Routine RecruitWUptake()  updates the site x depth level data structure site_hydr%recruit_w_uptake(j).  This is then applied to the uptake flux, here:
https://github.com/rgknox/fates/blob/fix-recruit-hydro/biogeophys/FatesPlantHydraulicsMod.F90#L2586-L2588

However, I think we could potentially improve the order of operations.  Since the recruit water uptake is already pre-determined, I think we should remove it from the soil water (site_hydr%h2osoi_liqvol_shell(:,:)) before implementing the hydraulic fluxes that follow.